src/page-objects/administration/ShippingListing.ts

```diff
@@ -0,0 +1,39 @@
+import type { Page, Locator } from '@playwright/test';
+import type { PageObject } from '../../types/PageObject';
+
+export class ShippingListing implements PageObject {
+    public readonly header: Locator;
+    public readonly addShippingMethod: Locator;
+    public readonly firstShippigMethodName: Locator;
+    public readonly firstShippingMethodContextButton: Locator;
+    public readonly contextMenu: Locator;
+    public readonly editButton: Locator;
+    public readonly deleteButton: Locator;
+
+    //warning modal
+    public readonly modal: Locator;
+    public readonly modalHeader: Locator;
+    public readonly modalCancelButton: Locator;
+    public readonly modalDeleteButton: Locator;
+
+    constructor(public readonly page: Page) {
+        this.header = page.locator('.smart-bar__header')
+        this.addShippingMethod = page.getByText('Add shipping method', { exact: true });
+        this.firstShippigMethodName = page.locator('.sw-data-grid__cell--name a').first();
+        this.firstShippingMethodContextButton = page.locator('.sw-data-grid__actions-menu').first();
```

Comment:
If you wanna use it in your test I don't know if that is so save, because the first item could change during your test if there a shipping method exists with a number in front of it. 

my recommendation would be the solution from @sguder  ...

```
async getLineItemByShippingMethodName(shippingMethodName: string): Promise<Record<string, Locator>> {
        const lineItem = this.page.locator('.sw-data-grid__row').filter({ hasText: shippingMethodName });
        const shippingMethodNameText = lineItem.locator('.sw-data-grid__cell--shipping-name');
        const contextMenuButton = lineItem.locator('.sw-data-grid-settings__trigger');

        return {
            shippingMethodNameText: shippingMethodNameText,
            contextMenuButton: contextMenuButton,
        }
```

Reply:
Actually I dont need them in my test, just I saw from the other pages similar location and would like to add them if need in the future. Like in FLowBuilderListing.ts . 
```
        this.firstFlowName = page.locator('.sw-data-grid__cell--name a').first();
        this.firstFlowContextButton = page.locator('.sw-data-grid__actions-menu').first();
```
But you are right, I removed them anyway maybe It could be flaky with multiple workers.